src/Modal.js

```diff
@@ -337,83 +231,215 @@ class Modal extends React.Component {
       onEntered,
       onExit,
       onExiting,
-      onExited: _,
-      onEntering: _1,
-      onEnter: _6,
-      onEntering: _4,
-      backdropClassName: _2,
+      onEnter,
+      onEntering,
+      onExited,
+      backdropClassName,
+      manager: propsManager,
       ...props
-    } = this.props;
+    },
+    ref,
+  ) => {
+    const [modalStyle, setStyle] = useState({});
+    const waitingForMouseUpRef = useRef(false);
+    const ignoreBackdropClickRef = useRef(false);
+
+    const [modal, setModalRef] = useCallbackRef();
+    const handleHide = useEventCallback(onHide);
+
+    bsPrefix = useBootstrapPrefix(bsPrefix, 'modal');
+
+    useImperativeHandle(
```

Comment:
should this expose anything else? like a DOM node or whatever?

Reply:
i don't think so not sure what help that would be, we can revisit tho if folks complain